public/app/core/history/RichHistoryLocalStorage.ts

```diff
@@ -173,7 +173,8 @@ function updateRichHistory(
  */
 function cleanUp(richHistory: RichHistoryLocalStorageDTO[]): RichHistoryLocalStorageDTO[] {
   const retentionPeriod: number = store.getObject(RICH_HISTORY_SETTING_KEYS.retentionPeriod, 7);
-  const retentionPeriodLastTs = createRetentionPeriodBoundary(retentionPeriod, false);
+  // We don't care about timezones that much here.
```

Comment:
Hmm not sure right now. Like we are cleaning up old things every time we add a new item. It probably does not make sense to clean it up differently based on the currently set timezone, like if you change for querying but it now deletes some additional history items it would be a bit weird. Like it probably should not be aligned to end of date at all and just do now - (14 x 24h) always. At the same time like does not seem like a big deal, not sure we even still use local history or not.

Reply:
Updated the comment though I hope that makes ti more clear.